cmd/vela-server/secret.go

```diff
@@ -51,5 +51,5 @@ func setupNative(c *cli.Context, d database.Service) (secret.Service, error) {
 // helper function to setup the Vault secret engine from the CLI arguments.
 func setupVault(c *cli.Context) (secret.Service, error) {
 	logrus.Tracef("Creating %s secret client from CLI configuration", constants.DriverVault)
-	return vault.New(c.String("vault-addr"), c.String("vault-token"), c.String("vault-version"), c.String("vault-prefix"))
+	return vault.New(c.String("vault-addr"), c.String("vault-token"), c.String("vault-version"), c.String("vault-prefix"), c.String("vault-auth-method"), c.String("vault-aws-role"), c.Duration("vault-renewal"))
```

Comment:
If we can think of a way to pass a struct vs values I think that would be more ideal. 

Values are good, just from a code readability standpoint, it feels better to pass a struct when you get into instances where functions have more than 4-5 parameters.

Reply:
It'd be great to come up with a standard for this in the Vela codebase 👍 

Something like:

> If a function takes more than 3 parameters of the same type, we should make it accept a struct instead.

And maybe we don't even care about the types of the parameters, and focus on the sheer number of parameters?

Reply:
Update to utilize a `vault.Config` struct